lib/hyrax/hirmeos/client.rb

```diff
@@ -18,25 +20,35 @@ def get_work(uuid)
     metrics_connection.get("/events?filter=work_uri:urn:uuid:#{uuid}") # This will need to be made configurable I think?
   end
 
-  def request_token
-    response = Faraday.post(URI.join(token_base_url, 'tokens'), { email: username, password: password }.to_json)
-    JSON.parse(response.body)['data'][0]['token']
+  def generate_token(payload)
```

Comment:
How about giving this a default value so it doesn't have to be passed in all the time?
```suggestion
  def generate_token(payload=build_payload)
```
Then the calls to this below could be simplified to just `generate_token`